### PR TITLE
Switch to golang native error wrapping

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This removes the direct `github.com/pkg/errors` dependency while using the native golang error wrapping.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Removed direct `github.com/pkg/errors` dependency by using golang native error wrapping.
```

### CODE REVIEW

File: mage/dependency.go [hunk 1]
```diff
@@ -44,15 +44,14 @@ func EnsureZeitgeist(version string) error {
 	}
 
 	if _, err := semver.ParseTolerant(version); err != nil {
-		return errors.Wrapf(
-			err,
-			"%s was not SemVer-compliant. Cannot continue.",
-			version,
+		return fmt.Errorf(
+			"%s was not SemVer-compliant. Cannot continue.: %w",
+			version, err,
```
Comment (xmudrii) [inline mage/dependency.go:49]: Nit: should we improve the formatting of this error? `Cannot continue.:` looks sort of strange.

File: command/command.go [hunk 2]
```diff
@@ -233,7 +233,7 @@ func (c *Command) RunSilentSuccessOutput() (output *Stream, err error) {
 		return nil, err
 	}
 	if !res.Success() {
-		return nil, errors.Errorf("command %v did not succeed: %v", c.String(), res.Error())
+		return nil, fmt.Errorf("command %v did not succeed: %v", c.String(), res.Error())
```
Comment (xmudrii) [inline command/command.go:236]: Why don't we use `%w` here?

File: mage/dependency.go (hunk 1 above)
Comment (saschagrunert) [reply mage/dependency.go]: Yep, changed to improve readability.

File: command/command.go (hunk 2 above)
Comment (saschagrunert) [reply command/command.go]: Exactly, good catch!

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**